src/message-io.js

```diff
@@ -83,47 +86,33 @@ module.exports = class MessageIO extends EventEmitter {
   startTls(credentialsDetails, hostname, trustServerCertificate) {
     const credentials = tls.createSecureContext ? tls.createSecureContext(credentialsDetails) : crypto.createCredentials(credentialsDetails);
 
-    this.securePair = tls.createSecurePair(credentials);
-    this.tlsNegotiationComplete = false;
-
-    this.securePair.on('secure', () => {
-      const cipher = this.securePair.cleartext.getCipher();
-
-      if (!trustServerCertificate) {
-        let verifyError = this.securePair.ssl.verifyError();
+    const duplexpair = new DuplexPair();
+    const securePair = this.securePair = {
+      cleartext: tls.connect({
+        socket: duplexpair.socket1,
+        secureContext: credentials,
+        rejectUnauthorized: !trustServerCertificate
+      }),
+      encrypted: duplexpair.socket2
+    };
 
-        // Verify that server's identity matches it's certificate's names
```

Comment:
By switching to `tls.connect`, we no longer need to perform server verification ourselves, and can instead make use of the `rejectUnauthorized` option. 🎉